scripts/west_commands/bindesc.py

```diff
@@ -0,0 +1,284 @@
+                                                     help='Search for a custom descriptor')
+        custom_search_parser.add_argument('type', type=str, choices=['UINT', 'STR'],
+                                          help='Descriptor type')
+        custom_search_parser.add_argument('id', type=str, help='Descriptor ID in hex')
+        custom_search_parser.add_argument('file', type=str, help='Executable file')
+        custom_search_parser.add_argument('--file-type', type=str, choices=self.EXTENSIONS,
+                                          help='File type')
+        custom_search_parser.add_argument('-b', '--big-endian', action='store_true',
+                                   help='Target CPU is big endian')
+        custom_search_parser.set_defaults(subcmd='custom_search', big_endian=False)
+
+        list_parser = subparsers.add_parser('list', help='List all known descriptors')
+        list_parser.set_defaults(subcmd='list', big_endian=False)
+
+        return parser
+
+    def dump(self, args):
+        image = self.get_image_data(args.file)
+
+        descriptors = self.parse_descriptors(image)
+        for tag, value in descriptors.items():
+            if tag in self.TAG_TO_NAME:
+                tag = self.TAG_TO_NAME[tag]
+            log.inf(f'{tag}', f'"{value}"' if isinstance(value, str) else f'{value}')
+
+    def list(self, args):
+        for tag in self.TAG_TO_NAME.values():
+            log.inf(f'{tag}')
+
+    def common_search(self, args, search_term):
+        image = self.get_image_data(args.file)
+
+        descriptors = self.parse_descriptors(image)
+
+        if search_term in descriptors:
+            value = descriptors[search_term]
+            log.inf(f'"{value}"' if isinstance(value, str) else f'{value}')
+        else:
+            log.die('Descriptor not found')
+
+    def search(self, args):
+        try:
+            search_term = self.NAME_TO_TAG[args.descriptor]
+        except KeyError:
+            log.die(f'Descriptor {args.descriptor} is invalid')
+
+        self.common_search(args, search_term)
+
+    def custom_search(self, args):
+        custom_type = {'STR': self.TYPE_STR, 'UINT': self.TYPE_UINT}[args.type]
+        custom_tag = self.bindesc_gen_tag(custom_type, int(args.id, 16))
+        self.common_search(args, custom_tag)
+
+    def do_run(self, args, _):
+        self.is_big_endian = args.big_endian
+        self.file_type = self.guess_file_type(args)
+        subcmd = getattr(self, args.subcmd)
+        subcmd(args)
+
+    def get_image_data(self, file_name):
+        if self.file_type == 'bin':
+            with open(file_name, 'rb') as bin_file:
+                return bin_file.read()
+
+        if self.file_type == 'hex':
+            return IntelHex(file_name).tobinstr()
+
+        if self.file_type == 'uf2':
+            with open(file_name, 'rb') as uf2_file:
+                return convert_from_uf2(uf2_file.read())
```

Comment:
Please rework this to avoid use of `assert` (use RuntimeError instead) and `print` (use `log`)